### PR TITLE
Make default block appender appear more consistently.

### LINKED ISSUE #5055: Inserter UI disappears when i type, and it's not obvious how to get them back
## Issue Overview

If you have no empty paragraph blocks, and you click out of editing the document so that is there no currently active block (e.g. click between the blocks and the right hand column of menus), the inserter and frequently used blocks are gone.

## Expected Behavior

If nothing is selected and you have no empty paragraph blocks, the inserter / frequently used blocks UI should still be visible, at the bottom of the document.

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
Fixes #5055. Also addresses some points raised in #13571 and #26404.

This PR makes the default block appender more consistent in when it appears.

### Current behavior in `master`

The appender appears after every block _except_ a Paragraph block. However, this has often been confusing and annoying in situations such as a Column where inserting a block after the last one in the Column feels weird because you have to do one of the following:

- Use the sibling inserter and then move the block down.
- Click the Paragraph, move your cursor to the end of the text, and press <kbd>Enter</kbd>.
- Click the Paragraph, and click the <kbd>Add block</kbd> button in the top bar.

None of these are particularly obvious at first. And the fact that the appender appears for every other block just makes it more confusing. I don't think most users will understand why a Heading or List at the end of a Column or post shows an appender, but a Paragraph doesn't.

### Behavior in this PR

The appender appears after every block including the Paragraph, with the exception that empty Paragraph blocks retain the current behavior in order to avoid having multiple appenders shown simultaneously.

I believe this behavior makes a whole lot more sense, and is a lot more predictable for users. I think there's definitely more work that could be done in the future (see #13571), but for now, this PR should greatly improve the situation.

## Screenshots 
image
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by ZebulanStanphill** [comment]
The `Writing Flow › Should navigate inner blocks with arrow keys` test failure is legit. At this point in the test, pressing the down arrow will not exit the Columns like you would expect (and like it does in `master`).

image

This is what happens instead:

image

It takes a total of 4 down arrow presses to get out of the Columns block when before it took just 1. Technically, though, this isn't a new problem. It was already happening with every other block... it just wasn't happening with the Paragraph block before this PR. So what should we do?

**Comment by jasmussen** [comment]
Thanks for your work.

High level, this seems acceptable for most cases. Paragraphs before / after:

before1

after1

Paragraphs in a group, before / after:

before2

after2

Even the consistency added to differing contents in columns seems worth the added consistency, despite the jump:

after2

The jump that happens in the default cover block behavior bugs me quite a bit, though. Before:

cover before

After:

cover after

I'm aware that it's an unfortunate configuration of the vertical alignment, and the serendipity of the paragraph not being followed by a sibling inserter by default (you have to press Enter like in text editors). But exactly because the Cover is so widely used, while consistent, it feels very jarring.

I know I've seen some separate conversations on how to solve this shift, for example by making that appender float, but the heuristic is a little tricky and I don't know if it's solvable easily. So I think we have to review this one on the assumption that this is how it's going to be. And for that reason, I'll need to sleep on it, but would love to hear from others on this! 

Thanks for your work.

**Comment by paaljoachim** [comment]
@shaunandrews I am thinking you might want to take a look at this PR.

File: packages/block-editor/src/components/default-block-appender/index.js [hunk 1]
```diff
@@ -75,22 +75,22 @@ export function DefaultBlockAppender( {
...
 		} = select( blockEditorStore );
 
 		const isEmpty = ! getBlockCount( ownProps.rootClientId );
-		const isLastBlockDefault =
+		const isLastBlockEmptyDefault =
 			getBlockName( ownProps.lastBlockClientId ) ===
-			getDefaultBlockName();
-		const isLastBlockValid = isBlockValid( ownProps.lastBlockClientId );
+				getDefaultBlockName() &&
+			getBlockAttributes( ownProps.lastBlockClientId )?.content === '';
```
Comment (ellatrix) [inline packages/block-editor/src/components/default-block-appender/index.js:89]: Should we use `isUnmodifiedDefaultBlock` here?

Comment (ellatrix) [reply packages/block-editor/src/components/default-block-appender/index.js:89]: (No guarantee that the default block has the content attribute and that it means the blocks is empty.)

**Comment by ZebulanStanphill** [comment]
Closing in favor of #36656.

**Pull request closed without merging**